snap/squashfs/squashfs.go

```diff
@@ -204,12 +204,16 @@ func (s *Snap) Unpack(src, dstDir string) error {
 
 	cmd := exec.Command("unsquashfs", "-n", "-f", "-d", dstDir, s.path, src)
 	cmd.Stderr = usw
-	if err := cmd.Run(); err != nil {
```

Comment:
@mvo do you remember if  the problem with unsquashfs such that it would only long certain errors and then still exit with 0 status? That's why the errors caught by usw were checked later. Perhaps this would be useful then:

```
stderr := &bytes.Buffer{}
cmd.Stderr = io.MultiWriter(&stderr, usw)
if err := cmd.Run(); err != nil {
	return osutil.OutputErr(stderr, err)
}
if usw.Err() != nil {
	return fmt.Errorf("cannot extract %q to %q: %v", src, dstDir, usw.Err())
}
```
(with the downside that if there's a lot of stderr output, the buffer will unfortunately grow significantly)

Reply:
I updated the code to follow this suggestion. I also explored a bit how a more modern unsquashfs behaves and it seems it stops after the first write failure (unlike the old versions) so the output is pretty short.